build_tools/ci/matmul_aiert_stability.sh

```diff
@@ -0,0 +1,576 @@
+       "'${IREE_INSTALL_DIR}', '${IREE_INSTALL_DIR}/bin', '${IREE_INSTALL_DIR}/tools'."
+  exit 1
+fi
+if [ -z "${TEST_RUNNER}" ]; then
+  echo "No 'iree-e2e-matmul-test' found in any of the following directories: " \
+       "'${IREE_INSTALL_DIR}', '${IREE_INSTALL_DIR}/bin', '${IREE_INSTALL_DIR}/tools'."
+  exit 1
+fi
+
+# Parameter 3) <peano-install-dir>
+if [ -z "${3-}" ]; then
+  PEANO=/opt/llvm-aie
+else
+  PEANO=`realpath "$3"`
+fi
+if [ ! -d "${PEANO}" ]; then
+  echo "No directory '${PEANO}' (argument 3) found."
+  exit 1
+fi
+
+# Parameter 4) <xrt-dir>
+if [ -z "${4-}" ]; then
+  XRT_DIR=/opt/xilinx/xrt
+else
+  XRT_DIR=`realpath "$4"`
+fi
+if [ ! -d "${XRT_DIR}" ]; then
+  echo "No directory '${XRT_DIR}' (argument 4) found."
+  exit 1
+fi
+
+# Parameter 5) <vitis-install-dir>
+if [ -z "${5-}" ]; then
+  VITIS=/opt/Xilinx/Vitis/2024.1
+else
+  VITIS=`realpath "$5"`
+fi
+if [ ! -d "${VITIS}" ]; then
+  echo "No directory '${VITIS}' (argument 5) found."
+  exit 1
+fi
+
+THIS_DIR="$(cd $(dirname $0) && pwd)"
+ROOT_DIR="$(cd $THIS_DIR/../.. && pwd)"
+
+GENERATOR="${ROOT_DIR}/tests/matmul/generate_e2e_matmul_tests.py"
+# Verify that generator exists
+if [ ! -f "${GENERATOR}" ]; then
+  echo "Generator script '${GENERATOR}' not found."
+  exit 1
+fi
+
+IREE_PYTHON3_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE:-python3}"
+if [ -z "$IREE_PYTHON3_EXECUTABLE" ]; then
+  echo "IREE_PYTHON3_EXECUTABLE is not set."
+  exit 1
+else
+  echo "Python version: $("${IREE_PYTHON3_EXECUTABLE}" --version)"
+fi
+
+source $XRT_DIR/setup.sh
+# Circumvent xclbin security (no longer needed as of April 2024 XDNA driver)
+export XRT_HACK_UNSECURE_LOADING_XCLBIN=1
+
+MM_KERNEL_URL=https://github.com/nod-ai/iree-amd-aie/releases/download/ukernels/mm.o
+ME_BASIC_URL=https://github.com/nod-ai/iree-amd-aie/releases/download/ukernels/me_basic.o
+
+if [ -d "$PEANO" ]; then
+  PEANO_ME_BASIC_FP="$PEANO/lib/aie2-none-unknown-elf/me_basic.o"
+  if [ -f "$PEANO_ME_BASIC_FP" ]; then
+    echo "File 'me_basic.o' already exists at $PEANO_ME_BASIC_FP"
+  else
+    echo "Downloading 'me_basic.o' to $PEANO_ME_BASIC_FP"
+    wget $ME_BASIC_URL -O "$PEANO_ME_BASIC_FP"
+  fi
+else
+  echo "Peano install not found at $PEANO; not downloading me_basic."
+fi
+
+if [ -f "${OUTPUT_DIR}/mm.o" ]; then
+  echo "File 'mm.o' already exists in ${OUTPUT_DIR}."
+else
+  echo "Downloading 'mm.o' to ${OUTPUT_DIR}/mm.o"
+  wget $MM_KERNEL_URL -O  "${OUTPUT_DIR}/mm.o"
+fi
+
+AIERT_COMMIT=$(git submodule status $ROOT_DIR/third_party/aie-rt | cut -d' ' -f2)
+
+if [ -x "${IREE_INSTALL_DIR}/bin/FileCheck" ]; then
+  FILECHECK_EXE="${IREE_INSTALL_DIR}/bin/FileCheck"
+elif [ -x "$(command -v FileCheck)" ]; then
+  FILECHECK_EXE="$(command -v FileCheck)"
+else
+  echo "FileCheck does not exist or isn't executable in ${IREE_INSTALL_DIR}/bin or on PATH."
+  exit 1
+fi
+
+VERBOSE=${VERBOSE:-0}
+DODIFF=${DODIFF:-0}
+GOLDEN_DIR=${GOLDEN_DIR:-$THIS_DIR/golden}
```

Comment:
is this on the CI machine, if thats the case then if something changes someone with no CI machine access cant fix it?

Reply:
Ah I see this script isnt actually run by CI, so its okay with me.